src/RegisterMapping.cpp

```diff
@@ -247,6 +251,37 @@ int8_t RegisterMapping::getBitIndex() const
     return m_bitIndex;
 }
 
+bool RegisterMapping::doesUpdate(const std::vector<uint16_t>& newValues)
+{
+    if (newValues.size() != m_byteValues.size())
+    {
+        throw std::logic_error("RegisterMapping: The value array has to be the same size, it cannot change.");
+    }
+
+    bool different = false;
+    bool significantChange = false;
+    uint32_t i = 0;
+    while (!different && i < newValues.size())
+    {
+        if (m_byteValues[i] != newValues[i])
+        {
+            different = true;
+            if (m_deadbandValue == 0.0)
+            {
+                significantChange = true;
+            }
+            else
+            {
+                significantChange = newValues[i] >= m_byteValues[i] + m_deadbandValue ||
+                                    newValues[i] <= m_byteValues[i] - m_deadbandValue;
+            }
+        }
+        ++i;
+    }
+
+    return (different || !m_isInitialized || !m_isValid) && significantChange;
```

Comment:
If it is initialized or valid after not being so then it should updated regardless of significantChange.

I think you should make this method being about checking the deadband and leave out those other validations (they are complicated enough even without duplicating them). Then call this from update as another condition